lib/_http_client.js

```diff
@@ -328,7 +328,7 @@ function ClientRequest(input, options, cb) {
 
   let optsWithoutSignal = options;
   if (optsWithoutSignal.signal) {
-    optsWithoutSignal = ObjectAssign({}, options);
+    optsWithoutSignal = ObjectAssign({ __proto__: null }, options);
     delete optsWithoutSignal.signal;
   }
```

Comment:
```suggestion
  const { signal, ...optsWithoutSignal } = options;
```

Reply:
I want to avoid code changes that are not related to this 😄